e2e/iam/helper_test.go

```diff
@@ -82,3 +84,127 @@ func deleteOrgAPIKey(id string) error {
 	cmd.Env = os.Environ()
 	return cmd.Run()
 }
+
+func createProject(projectName string) (string, error) {
+	cliPath, err := e2e.Bin()
+	if err != nil {
+		return "", err
+	}
+	cmd := exec.Command(cliPath,
+		iamEntity,
+		projectEntity,
+		"create",
+		projectName,
+		"-o=json")
+	cmd.Env = os.Environ()
+	resp, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+
+	var project mongodbatlas.Project
+	if err := json.Unmarshal(resp, &project); err != nil {
+		return "", err
+	}
+
+	if project.Name != projectName {
+		return "", fmt.Errorf(`got="%s" want="%s"`, project.Name, projectName)
+	}
+	return project.ID, nil
+}
+
+func deleteProject(projectID string) error {
+	cliPath, err := e2e.Bin()
+	if err != nil {
+		return err
+	}
+	cmd := exec.Command(cliPath,
+		iamEntity,
+		projectEntity,
+		"delete",
+		projectID,
+		"--force")
+	cmd.Env = os.Environ()
+	return cmd.Run()
+}
+
+func createTeam(teamName string) (string, error) {
+	cliPath, err := e2e.Bin()
+	if err != nil {
+		return "", err
+	}
+	username, _, err := getUserFromOrg(0)
+
+	if err != nil {
+		return "", err
+	}
+	cmd := exec.Command(cliPath,
+		iamEntity,
+		teamsEntity,
+		"create",
+		teamName,
+		"--username",
+		username,
+		"-o=json")
+	cmd.Env = os.Environ()
+	resp, err := cmd.CombinedOutput()
+
+	if err != nil {
+		return "", err
+	}
+
+	var team mongodbatlas.Team
+	if err := json.Unmarshal(resp, &team); err != nil {
+		return "", err
+	}
+
+	return team.ID, nil
+}
+
+func deleteTeam(teamID string) error {
+	cliPath, err := e2e.Bin()
+	if err != nil {
+		return err
+	}
+	cmd := exec.Command(cliPath,
+		iamEntity,
+		teamsEntity,
+		"delete",
+		teamID,
+		"--force")
+	cmd.Env = os.Environ()
+	return cmd.Run()
+}
+
+// getUserFromOrg returns the user at the position userIndex.
+// We need to pass the userIndex because the command iam teams users add would not work
+// if the the user is already in the team.
+func getUserFromOrg(userIndex int) (username, userID string, err error) {
+	cliPath, err := e2e.Bin()
+	if err != nil {
+		return "", "", err
+	}
+	cmd := exec.Command(cliPath,
+		iamEntity,
+		orgEntity,
+		usersEntity,
+		"list",
+		"-o=json")
+	cmd.Env = os.Environ()
+	resp, err := cmd.CombinedOutput()
+
+	if err != nil {
+		return "", "", err
+	}
+
+	var users mongodbatlas.AtlasUsersResponse
+	if err := json.Unmarshal(resp, &users); err != nil {
+		return "", "", err
+	}
+
+	if len(users.Results) < userIndex {
+		return "", "", fmt.Errorf(`got="%s" want= len(users.Results) > "%d"`, users.Results, userIndex)
```

Comment:
yes on the error but the error message can be better 
```suggestion
		return "", "", fmt.Errorf("invalid index %d for %d users", n, len(users.Results))
```